app/controllers/concerns/effective/crud_controller.rb

```diff
@@ -21,7 +21,17 @@ module ClassMethods
       # This is used for the buttons/submits/ons
       # It doesn't really work with the resource_scope correctly but the routes are important here
       def effective_resource
-        @_effective_resource ||= Effective::Resource.new(controller_path)
+        @_effective_resource ||= begin
+          controller = new()
+          klass = (controller.resource_scope_relation.call().try(:klass) rescue false) if controller.respond_to?(:resource_scope_relation)
+
+          if klass.present?
+            namespace = controller_path.split('/')[0...-1].join('/').presence
+            Effective::Resource.new(klass, namespace: namespace)
+          else
+            Effective::Resource.new(controller_path)
+          end
+        end
```

Comment:
This probably isn't correct.

But is a huge rabbit hole of rewriting how the effective controller does the 'submits, buttons and ons'